### PR TITLE
New weights

### PR DESCRIPTION
**Pull request by NachoPal**

### CODE REVIEW

File: polkadot-parachains/statemine/src/weights/frame_system.rs [hunk 1]
```diff
@@ -45,42 +44,40 @@ use sp_std::marker::PhantomData;
...
 	fn set_storage(i: u32, ) -> Weight {
 		(0 as Weight)
 			// Standard Error: 0
-			.saturating_add((737_000 as Weight).saturating_mul(i as Weight))
+			.saturating_add((517_000 as Weight).saturating_mul(i as Weight))
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(i as Weight)))
 	}
 	// Storage: Skipped Metadata (r:0 w:0)
 	fn kill_storage(i: u32, ) -> Weight {
-		(1_312_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/frame_system.rs:74]: very big drift?

File: polkadot-parachains/statemine/src/weights/frame_system.rs [hunk 2]
```diff
@@ -45,42 +44,40 @@ use sp_std::marker::PhantomData;
...
+		(0 as Weight)
 			// Standard Error: 0
-			.saturating_add((496_000 as Weight).saturating_mul(i as Weight))
+			.saturating_add((372_000 as Weight).saturating_mul(i as Weight))
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(i as Weight)))
 	}
 	// Storage: Skipped Metadata (r:0 w:0)
 	fn kill_prefix(p: u32, ) -> Weight {
-		(3_976_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/frame_system.rs:81]: big drift

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs [hunk 3]
```diff
@@ -47,13 +46,13 @@ pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
 	// Storage: Assets Asset (r:1 w:1)
 	fn create() -> Weight {
-		(36_997_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/pallet_assets.rs:50]: big drift

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs [hunk 4]
```diff
@@ -82,122 +81,120 @@ impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
...
-		(49_191_000 as Weight)
+		(35_699_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	// Storage: Assets Asset (r:1 w:1)
 	// Storage: Assets Account (r:2 w:2)
 	// Storage: System Account (r:1 w:1)
 	fn transfer() -> Weight {
-		(72_470_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/pallet_assets.rs:100]: big drift

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs [hunk 5]
```diff
@@ -82,122 +81,120 @@ impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
...
-		(61_123_000 as Weight)
+		(44_544_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(4 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
 	// Storage: Assets Asset (r:1 w:1)
 	// Storage: Assets Account (r:2 w:2)
 	// Storage: System Account (r:1 w:1)
 	fn force_transfer() -> Weight {
-		(72_777_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/pallet_assets.rs:116]: big drift

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs [hunk 6]
```diff
@@ -82,122 +81,120 @@ impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
...
 	fn thaw_asset() -> Weight {
-		(22_168_000 as Weight)
+		(16_576_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: Assets Asset (r:1 w:1)
 	// Storage: Assets Metadata (r:1 w:0)
 	fn transfer_ownership() -> Weight {
-		(24_868_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/pallet_assets.rs:149]: big drift

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs [hunk 7]
```diff
@@ -206,21 +203,21 @@ impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
 	// Storage: Assets Account (r:2 w:2)
 	// Storage: System Account (r:1 w:1)
 	fn transfer_approved() -> Weight {
-		(94_204_000 as Weight)
```
Comment (coderobe) [inline polkadot-parachains/statemine/src/weights/pallet_assets.rs:209]: big drift

**Review by coderobe — COMMENTED** [review]
Due to the recent DB cache size increase (128 -> 1024) many of these drift quite a bit, usually downwards

**Comment by coderobe** [comment]
I've only highlighted some of the bigger changes, but you get the gist of it - the recent DB cache size change has improved perf and thus lowered the weights across the board

**Comment by shawntabrizi** [comment]
> Due to the recent DB cache size increase (128 -> 1024) many of these drift quite a bit, usually downwards

This change will not affect the weights you are pointing out at all.

**Comment by gilescope** [comment]
DB CACHE: 128 -> 1024 - is this the reason why the weights in general are mostly 30% reduced?

**Review by gilescope — APPROVED** [review]
Seems in line changes in general. As long as we are ok with 0 weight for remarks then looks reasonable. (remark_with_event is already 0 weighted)

File: polkadot-parachains/westmint/src/weights/pallet_multisig.rs [hunk 8]
```diff
@@ -98,48 +97,48 @@ impl<T: frame_system::Config> pallet_multisig::WeightInfo for WeightInfo<T> {
...
+			.saturating_add((140_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: Multisig Multisigs (r:1 w:1)
 	// Storage: Multisig Calls (r:1 w:1)
 	// Storage: System Account (r:1 w:1)
 	fn approve_as_multi_complete(s: u32, ) -> Weight {
-		(101_474_000 as Weight)
-			// Standard Error: 0
```
[the remark this answers is not in the document]
Comment (gilescope) [reply polkadot-parachains/westmint/src/weights/pallet_multisig.rs:132]: (Didn't go down much last time but did a 1/3 off this time.)

**Comment by apopiak** [comment]
just FYI: the Substrate weights went down when running the benchmarks with optimizations: https://github.com/paritytech/substrate/pull/10692

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs [hunk 9]
```diff
@@ -82,122 +81,120 @@ impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
...
+		(32_058_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: Assets Asset (r:1 w:0)
 	// Storage: Assets Metadata (r:1 w:1)
-	fn force_set_metadata(_n: u32, s: u32, ) -> Weight {
-		(23_906_000 as Weight)
-			// Standard Error: 0
-			.saturating_add((4_000 as Weight).saturating_mul(s as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/statemine/src/weights/pallet_assets.rs:180]: `force_set_metadata` does not depend on parameter `s` anymore. This is one of the important things to spot @coderobe @gilescope

File: polkadot-parachains/statemint/src/weights/pallet_assets.rs [hunk 10]
```diff
@@ -82,122 +81,124 @@ impl<T: frame_system::Config> pallet_assets::WeightInfo for WeightInfo<T> {
...
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: Assets Asset (r:1 w:0)
 	// Storage: Assets Metadata (r:1 w:1)
-	fn force_set_metadata(_n: u32, s: u32, ) -> Weight {
-		(22_035_000 as Weight)
+	fn force_set_metadata(n: u32, s: u32, ) -> Weight {
+		(19_698_000 as Weight)
+			// Standard Error: 0
+			.saturating_add((1_000 as Weight).saturating_mul(n as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/statemint/src/weights/pallet_assets.rs:179]: `force_set_metadata` suddenly depends on `n` @coderobe @gilescope

File: polkadot-parachains/statemint/src/weights/pallet_proxy.rs [hunk 11]
```diff
@@ -47,95 +46,93 @@ pub struct WeightInfo<T>(PhantomData<T>);
...
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: unknown [0x3a65787472696e7369635f696e646578] (r:1 w:0)
 	// Storage: Proxy Proxies (r:1 w:1)
-	fn anonymous(p: u32, ) -> Weight {
-		(43_815_000 as Weight)
-			// Standard Error: 3_000
-			.saturating_add((20_000 as Weight).saturating_mul(p as Weight))
+	fn anonymous(_p: u32, ) -> Weight {
```
Comment (NachoPal) [inline polkadot-parachains/statemint/src/weights/pallet_proxy.rs:126]: It does not depend on `p` anymore @coderobe @gilescope

File: polkadot-parachains/statemine/src/weights/pallet_uniques.rs [hunk 12]
```diff
@@ -46,34 +45,37 @@ use sp_std::marker::PhantomData;
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_uniques::WeightInfo for WeightInfo<T> {
 	// Storage: Uniques Class (r:1 w:1)
+	// Storage: Uniques ClassAccount (r:0 w:1)
 	fn create() -> Weight {
-		(37_882_000 as Weight)
+		(31_145_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/statemine/src/weights/pallet_uniques.rs:52]: There is an extra write @coderobe @gilescope

File: polkadot-parachains/statemine/src/weights/pallet_uniques.rs [hunk 13]
```diff
@@ -46,34 +45,37 @@ use sp_std::marker::PhantomData;
...
 	}
 	// Storage: Uniques Class (r:1 w:1)
+	// Storage: Uniques ClassAccount (r:0 w:1)
 	fn force_create() -> Weight {
-		(19_568_000 as Weight)
+		(17_647_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/statemine/src/weights/pallet_uniques.rs:59]: There is an extra write @coderobe @gilescope

File: polkadot-parachains/statemint/src/weights/pallet_uniques.rs [hunk 14]
```diff
@@ -46,34 +45,37 @@ use sp_std::marker::PhantomData;
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_uniques::WeightInfo for WeightInfo<T> {
 	// Storage: Uniques Class (r:1 w:1)
+	// Storage: Uniques ClassAccount (r:0 w:1)
 	fn create() -> Weight {
-		(36_962_000 as Weight)
+		(31_395_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/statemint/src/weights/pallet_uniques.rs:52]: There is an extra write

File: polkadot-parachains/statemint/src/weights/pallet_uniques.rs [hunk 15]
```diff
@@ -46,34 +45,37 @@ use sp_std::marker::PhantomData;
...
 	}
 	// Storage: Uniques Class (r:1 w:1)
+	// Storage: Uniques ClassAccount (r:0 w:1)
 	fn force_create() -> Weight {
-		(19_734_000 as Weight)
+		(18_120_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/statemint/src/weights/pallet_uniques.rs:59]: There is an extra write

File: polkadot-parachains/westmint/src/weights/pallet_uniques.rs [hunk 16]
```diff
@@ -46,34 +45,37 @@ use sp_std::marker::PhantomData;
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_uniques::WeightInfo for WeightInfo<T> {
 	// Storage: Uniques Class (r:1 w:1)
+	// Storage: Uniques ClassAccount (r:0 w:1)
 	fn create() -> Weight {
-		(38_475_000 as Weight)
+		(31_101_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/westmint/src/weights/pallet_uniques.rs:52]: There is an extra write

File: polkadot-parachains/westmint/src/weights/pallet_uniques.rs [hunk 17]
```diff
@@ -46,34 +45,37 @@ use sp_std::marker::PhantomData;
...
 	}
 	// Storage: Uniques Class (r:1 w:1)
+	// Storage: Uniques ClassAccount (r:0 w:1)
 	fn force_create() -> Weight {
-		(19_667_000 as Weight)
+		(17_191_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
```
Comment (NachoPal) [inline polkadot-parachains/westmint/src/weights/pallet_uniques.rs:59]: There is an extra write

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs (hunk 9 above)
Comment (gilescope) [reply polkadot-parachains/statemine/src/weights/pallet_assets.rs:180]: I see. Why is that?

File: polkadot-parachains/statemine/src/weights/pallet_uniques.rs (hunk 12 above)
Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_uniques.rs:52]: Aligned with Substrate weights: https://github.com/paritytech/substrate/blob/polkadot-v0.9.16/frame/uniques/src/weights.rs#L80

File: polkadot-parachains/statemine/src/weights/pallet_uniques.rs (hunk 13 above)
Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_uniques.rs:59]: Aligned with Substrate weights: https://github.com/paritytech/substrate/blob/polkadot-v0.9.16/frame/uniques/src/weights.rs#L80

File: polkadot-parachains/statemint/src/weights/pallet_uniques.rs (hunk 14 above)
Comment (NachoPal) [reply polkadot-parachains/statemint/src/weights/pallet_uniques.rs:52]: Aligned with Substrate weights

File: polkadot-parachains/statemint/src/weights/pallet_uniques.rs (hunk 15 above)
Comment (NachoPal) [reply polkadot-parachains/statemint/src/weights/pallet_uniques.rs:59]: Aligned with Substrate weights

File: polkadot-parachains/westmint/src/weights/pallet_uniques.rs (hunk 16 above)
Comment (NachoPal) [reply polkadot-parachains/westmint/src/weights/pallet_uniques.rs:52]: Aligned with Substrate weights

File: polkadot-parachains/westmint/src/weights/pallet_uniques.rs (hunk 17 above)
Comment (NachoPal) [reply polkadot-parachains/westmint/src/weights/pallet_uniques.rs:59]: Aligned with Substrate weights

File: polkadot-parachains/statemine/src/weights/pallet_assets.rs (hunk 9 above)
Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_assets.rs:180]: I don't know, it does not match with Substrate weights, we have to find out

Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_assets.rs:180]: Actually, according to this: https://github.com/paritytech/substrate/blob/polkadot-v0.9.16/frame/assets/src/lib.rs#L1036 it should depend on both

File: polkadot-parachains/statemint/src/weights/pallet_assets.rs (hunk 10 above)
Comment (NachoPal) [reply polkadot-parachains/statemint/src/weights/pallet_assets.rs:179]: This is correct though and aligned with substrate

Comment (NachoPal) [reply polkadot-parachains/statemint/src/weights/pallet_assets.rs:179]: However, in their weights `n` is ignored.

File: polkadot-parachains/statemint/src/weights/frame_system.rs [hunk 18]
```diff
@@ -45,42 +44,40 @@ use sp_std::marker::PhantomData;
 /// Weight functions for `frame_system`.
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> frame_system::WeightInfo for WeightInfo<T> {
-	fn remark(b: u32, ) -> Weight {
-		(814_000 as Weight)
-			// Standard Error: 0
-			.saturating_add((1_000 as Weight).saturating_mul(b as Weight))
+	fn remark(_b: u32, ) -> Weight {
+		(0 as Weight)
```
[the remark this answers is not in the document]
Comment (ggwpez) [reply polkadot-parachains/statemint/src/weights/frame_system.rs:48]: There is still the per-byte-fee.  
We talked about it here: https://github.com/paritytech/substrate/pull/10692#discussion_r789746400

**Pull request merged**